File: Software Technologies/05. ExpressJS Calculator/models/Calculator.js
function Calculator(leftOperand, operator, rightOperand) {
    this.leftOperand = leftOperand;
    this.operator = operator;
    this.rightOperand = rightOperand;

    this.calculateResult = function () {
        let result;
        if (!isNaN(this.leftOperand) && !isNaN(this.rightOperand))
            switch (this.operator) {
                case '+':
                    result = this.leftOperand + this.rightOperand;
                    break;
                case '-':
                    result = this.leftOperand - this.rightOperand;
                    break;
                case '*':
                    result = this.leftOperand * this.rightOperand;
                    break;
                case '/':
                    result = this.leftOperand / this.rightOperand;
                    break;
                case '%':
                    result = this.leftOperand % this.rightOperand;
                    break;
                case '^':
                    result = this.leftOperand ** this.rightOperand;
                    break;
            }
        return result;
    }
}

module.exports = Calculator;

File: Software Technologies/16. ASP.NET - Exercises/01. Calculator/Calculator-CSharp/Models/Calculator.cs
namespace Calculator_CSharp.Models
{
    public class Calculator
    {
        public decimal LeftOperand { get; set; }
        public decimal RightOperand { get; set; }
        public string Operator { get; set; }
        public decimal Result { get; set; }

        public Calculator()
        {
            Result = 0;
        }
    }
}
